engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnDecisionBehavior.java

```diff
@@ -215,11 +218,24 @@ private void writeDecisionEvaluationEvent(
             .setElementInstanceKey(context.getElementInstanceKey())
             .setElementId(context.getElementId());
 
+    final var decisionKeysByDecisionId =
+        decisionState
+            .findDecisionsByDecisionRequirementsKey(decision.getDecisionRequirementsKey())
+            .stream()
+            .collect(
+                Collectors.toMap(
+                    persistedDecision -> bufferAsString(persistedDecision.getDecisionId()),
+                    PersistedDecision::getDecisionKey));
+
     decisionResult
         .getEvaluatedDecisions()
         .forEach(
             evaluatedDecision ->
-                addDecisionToEvaluationEvent(evaluatedDecision, decisionEvaluationEvent));
+                addDecisionToEvaluationEvent(
+                    evaluatedDecision,
+                    decisionKeysByDecisionId.getOrDefault(
+                        evaluatedDecision.decisionId(), UNKNOWN_DECISION_KEY),
```

Comment:
💭 I guess the decisionKeysBuDecisionId should always have the key for the evaluated decision. If not, we probably have a consistency issue. 

Ping @oleschoenburg, not sure if this is really useful or whether you have a plan to deal with such inconsistencies, but just wanted to bring this one to your attention.